maps/src/G3SkyMap.cxx

```diff
@@ -32,13 +36,23 @@ G3SkyMap::serialize(A &ar, unsigned v)
 		} else {
 			overflow = 0;
 		}
-		init_from_v1_data(dims, dat);
+		InitFromV1Data(dims, dat);
 	} else {
 		ar & cereal::make_nvp("overflow", overflow);
 	}
 
 	ar & cereal::make_nvp("pol_type", pol_type);
-	ar & cereal::make_nvp("is_weighted", is_weighted);
+	ar & cereal::make_nvp("weighted", weighted);
+
+	if (v > 2)
+		ar & cereal::make_nvp("pol_conv", pol_conv);
+	else
+		pol_conv = ConvNone;
+
+	if (pol_type == U && pol_conv == ConvNone)
+		log_warn("Map object has pol_type U and unknown pol_conv. "
+			 "Set the pol_conv attribute to IAU or COSMO.");
```

Comment:
I worry this will be confusing. Serialization and deserialization can happen at non-obvious times, including implicit times (for example, when using an MPI communicator), and it may not be clear from the message what is happening or why. Is this check only for V1 and V2 maps? If so, this should be gated on the version number and the message should probably say something about it being old data. Otherwise, the setter is probably the right place to warn.

Reply:
This check is now only done on V1 and V2 maps, and only when the attribute is explicitly set to None for U maps.  The code that requires a polarization convention will rely on the attribute being set appropriately, as in, e.g. the FlattenPol function.  See cb3c3af